pilot/pkg/kube/inject/app_probe.go

```diff
@@ -74,30 +77,41 @@ func rewriteAppHTTPProbe(spec *SidecarInjectionSpec, podSpec *corev1.PodSpec) {
 				// Matches the regex pattern, but without actual values provided.
 				if len(c.Args) <= i+1 {
 					log.Errorf("No statusPort value provided, skip app probe rewriting")
-					return
+					return -1
 				}
 				portStr = c.Args[i+1]
 			}
 			p, err := strconv.Atoi(portStr)
 			if err != nil {
 				log.Errorf("Failed to convert statusPort to int %v, err %v", portStr, err)
-				return
+				return -1
 			}
 			statusPort = p
 			break
 		}
 	}
+	return statusPort
+}
+
+func calculateProbeRewrite(podSpec *corev1.PodSpec, spec *SidecarInjectionSpec, modify bool) []rfc6902PatchOperation {
+	patch := []rfc6902PatchOperation{}
+	if spec == nil || podSpec == nil {
+		return patch
+	}
+	statusPort := extractStatusPort(spec)
 	// Pilot agent statusPort is not defined, skip changing application http probe.
 	if statusPort == -1 {
-		return
+		return patch
 	}
 
 	// Change the application containers' probe to point to sidecar's status port.
-	rewriteProbe := func(probe *corev1.Probe, portMap map[string]int32) {
+	rewriteProbe := func(probe *corev1.Probe, portMap map[string]int32, path string) *rfc6902PatchOperation {
 		if probe == nil || probe.HTTPGet == nil {
-			return
+			return nil
 		}
-		httpGet := probe.HTTPGet
+		httpGet := proto.Clone(probe.HTTPGet).(*corev1.HTTPGetAction)
+		// Walkaround... proto.Clone can't copy corev1.IntOrStr somehow...
```

Comment:
The previous code didn't have this? Was that a bug?

Also, in the comment I think it should be "workaround" not "walkaround".

Reply:
aha... changed.

previous code we modify the probe proto struct directly so no need to clone and doesn't have this problem.